### PR TITLE
feature: add CPU and Memory pool BDCom Switchs

### PR DESCRIPTION
**Pull request by kakopedreros**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #6522

### CODE REVIEW

File: includes/discovery/mempools/bdcom.inc.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+
+ * @package    LibreNMS
+ * @subpackage discovery
+ * @link       http://librenms.org
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.48.1 nmsMemoryPoolUtilization.0 = [%][This is the memory pool utilization"]
```
Comment (laf) [inline includes/discovery/mempools/bdcom.inc.php]: Please remove all the comments here, this is all in the mib `mibs/pbn/NMS-MEMORY-POOL-MIB.MIB`

File: includes/discovery/mempools/bdcom.inc.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ .1.3.6.1.4.1.3320.9.48.7 nmsMemoryPoolMessageBufferRatio.0 = [%]["This is the ratio of message buffer size to total memory size"]
+ .1.3.6.1.4.1.3320.9.48.8 nmsMemoryPoolMessageBufferUtilization.0 = [%]["This is the message buffer of utilization"]
+ .1.3.6.1.4.1.3320.9.48.9 nmsMemoryPoolTotalFlashSize.0 = [Unsigned32][nmsMemoryPoolTotalFlashSize]
+
+ */
+
+if ($device['os'] == 'bdcom') {
+    echo 'BDCOM, NMS-MEMORY-POOL:';
+
+    $total = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
Comment (laf) [inline includes/discovery/mempools/bdcom.inc.php]: please use the named OID as we have the MIB available. It makes it easier to read

Comment (laf) [inline includes/discovery/mempools/bdcom.inc.php]: In fact you can also use snmp_get_multi or snmp_get_multi_oid in place of two gets.

File: includes/discovery/processors/bdcom.inc.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+
+ * @package    LibreNMS
+ * @subpackage discovery
+ * @link       http://librenms.org
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.109.1.1.1.1.1.1 nmspmCPUTotalIndex.1 = [Unsigned32][Index]
```
Comment (laf) [inline includes/discovery/processors/bdcom.inc.php]: Same comments for the comment data you have in here + using the named OID.

File: includes/polling/mempools/bdcom.inc.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+
+ * @package    LibreNMS
+ * @subpackage polling
+ * @link       http://librenms.org
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.48.1 nmsMemoryPoolUtilization.0 = [%][This is the memory pool utilization"]
```
Comment (laf) [inline includes/polling/mempools/bdcom.inc.php]: Please remove comments again.

File: includes/polling/mempools/bdcom.inc.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ .1.3.6.1.4.1.3320.9.48.4 nmsMemoryPoolRegionRatio.0 = [%]["This is the ratio of total region size to total memory size"]
+ .1.3.6.1.4.1.3320.9.48.5 nmsMemoryPoolHeapRatio.0 = [%]["This is the ratio of heap size to total memory size"]
+ .1.3.6.1.4.1.3320.9.48.6 nmsMemoryPoolHeapUtilization.0 = [%]["This is the heap utilization"]
+ .1.3.6.1.4.1.3320.9.48.7 nmsMemoryPoolMessageBufferRatio.0 = [%]["This is the ratio of message buffer size to total memory size"]
+ .1.3.6.1.4.1.3320.9.48.8 nmsMemoryPoolMessageBufferUtilization.0 = [%]["This is the message buffer of utilization"]
+ .1.3.6.1.4.1.3320.9.48.9 nmsMemoryPoolTotalFlashSize.0 = [Unsigned32][nmsMemoryPoolTotalFlashSize]
+
+ */
+
+$total = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
Comment (laf) [inline includes/polling/mempools/bdcom.inc.php]: Use named OIDs + multi get

File: includes/polling/mempools/bdcom.inc.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+$used  = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.1.0', '-OvQ');
+
+echo 'BDCOM Memory Pool';
+
+$mempool['total'] = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
Comment (laf) [inline includes/polling/mempools/bdcom.inc.php]: You seemed to have duplicated the next two snmp calls.

File: includes/discovery/mempools/bdcom.inc.php (hunk 1 above)
Comment (kakopedreros) [inline includes/discovery/mempools/bdcom.inc.php]: All the comments, including the license and author info?

File: includes/discovery/processors/bdcom.inc.php (hunk 3 above)
Comment (kakopedreros) [inline includes/discovery/processors/bdcom.inc.php]: if i use the named OID, i need to declare the mib_dir in the definition os file?

File: includes/discovery/mempools/bdcom.inc.php (hunk 1 above)
Comment (laf) [inline includes/discovery/mempools/bdcom.inc.php]: No, license and author are fine.

File: includes/discovery/processors/bdcom.inc.php (hunk 3 above)
Comment (laf) [inline includes/discovery/processors/bdcom.inc.php]: Only if the mib dir is named differently from the os name, which it should if the vendor has multiple products so the mibs go in the vendor dir then.

File: includes/discovery/mempools/bdcom.inc.php (hunk 2 above)
Comment (kakopedreros) [inline includes/discovery/mempools/bdcom.inc.php]: Using the named OID i need to declare de MIB, i wich way?

This...
`snmp_get($device, 'nmsMemoryPoolTotalMemorySize.0', '-Ovq', 'NMS-MEMORY-POOL');`

or this one...?
`snmp_get($device, 'NMS-MEMORY-POOL::nmsMemoryPoolTotalMemorySize.0', '-Ovq');`

File: includes/polling/mempools/bdcom.inc.php (hunk 4 above)
Comment (kakopedreros) [inline includes/polling/mempools/bdcom.inc.php]: done.

File: includes/polling/mempools/bdcom.inc.php (hunk 6 above)
Comment (kakopedreros) [inline includes/polling/mempools/bdcom.inc.php]: done.

File: includes/discovery/mempools/bdcom.inc.php (hunk 1 above)
Comment (kakopedreros) [inline includes/discovery/mempools/bdcom.inc.php]: done.

**Comment by laf** [comment]
@kakopedreros I'm not seeing any of my requests having been changed yet, let us know if you are having issues.

**Comment by kakopedreros** [comment]
I having problems with the named OID, with the mib files located in mibs/pbn folder doesnt work.
Maybe it is the mib files in this folder are incomplete or the MIB files from BDCom are in some way differents.

[bdcom_nms_mibs.txt](https://github.com/librenms/librenms/files/968988/bdcom_nms_mibs.txt)


What do you prefere, upload the NMs mibs provided by BDCom or only use numeric OID?

**Comment by laf** [comment]
You can upload the mib as well, it's extremely useful and easier to read by using mibs.

**Comment by laf** [comment]
@thanks for updating, I'll take a proper look later on but one thing I spotted was the OS logo has to be 32x32 if it's a png, can you please revert that unless you can swap it out to svg

**Comment by kakopedreros** [comment]
Replaced the png logo with a new logo in svg. Modified the processors discovery module. Detected an error in the variable type detection.

pre-commit.php: sucess

**Comment by kakopedreros** [comment]
one more thing...

This was tested in 136 BDCom switches... =)
captura de pantalla 2017-05-02 a la s 14 53 20

**Comment by laf** [comment]
I've just compressed the svg down so will be good after that.

**Pull request merged**